pkg/engine/api/registryclientfactory.go

```diff
@@ -0,0 +1,184 @@
+package api
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	kyvernov1 "github.com/kyverno/kyverno/api/kyverno/v1"
+	"github.com/kyverno/kyverno/pkg/config"
+	"github.com/kyverno/kyverno/pkg/registryclient"
+
+	kubeinformers "k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	resyncPeriod = 15 * time.Minute
+)
+
+type RegistryClientFactory interface {
+	GetClient(ctx context.Context, creds *kyvernov1.ImageRegistryCredentials) (registryclient.Client, error)
+}
+
+type registryClientFactory struct {
+	globalClient registryclient.Client
+	kubeClient   kubernetes.Interface
```

Comment:
we probably don't want a kube client here.

Reply:
ah yes, got it!

Reply:
fixed